engine/engine_test.go

```diff
@@ -1223,7 +1223,7 @@ func TestQueriesAgainstOldEngine(t *testing.T) {
 			name: "scalar func with NaN",
 			load: `load 30s
 		 	http_requests_total{pod="nginx-1"} 1+1x15
-		 	http_requests_total{pod="nginx-2"} 1+2x18`,
+		 	http_requests_total{pod="nginx-2"} NaN`,
```

Comment:
I randomly added this, since the name of the function implies that its to be tested with a NaN value